hm_pyhelper/diagnostics/diagnostics_report.py

```diff
@@ -33,10 +33,33 @@ def __init__(self, diagnostics=[], **kwargs):
 
         if DIAGNOSTICS_ERRORS_KEY not in self:
             self.__setitem__(DIAGNOSTICS_ERRORS_KEY, [])
+
         self.diagnostics = diagnostics
 
+        # Needed for more verbose messages at HPT end. This attribute gets
+        # queried to display exactly which keys failed causing the diagnostics
+        # to fail.
+        self.manufacturing_errors = []
+
     def passed(self):
-        return self[DIAGNOSTICS_PASSED_KEY]
+        # Check only a subset of keys that are required for manufacturing
+        # tests. If these don't have errors then we can conclude that device
+        # is good from manufacturing point-of-view.
+        keys_to_check = {
+            'ECC', 'OK', 'E0', 'W0', 'PK', 'BT', 'VA', 'FR', 'serial_number'
+        }
+
+        errors = self.get_errors()
+        if errors:
+            self.set_passed(False)
+            self.manufacturing_errors = keys_to_check.intersection(errors)
+            if len(self.manufacturing_errors) > 0:
+                return False
```

Comment:
Clients may want to know which keys failed. Maybe we should rename the method `get_errors` and return `manufacturing_errors` instead of true/false.

Reply:
Are we using DiagnosticsReport in places other than in Hotspot Production Tool (like hm-diag)? If that's the case then we might need to change those repos too. Otherwise I think adding `get_manufacturing_errors` or `has_manufacturing_errors` in addition to `get_errors` should be fine?